tests/rtdl/rtld_next/meson.build

```diff
@@ -0,0 +1,25 @@
+# Prevent tail calls, as it breaks libc's 'calling object' detection.
+no_tail_calls = '-fno-optimize-sibling-calls'
```

Comment:
Is this needed despite `[[gnu::noinline]]`?

Reply:
Yes:
```c
void *foo() {
    return dlsym(...);
}
```
Even though dlsym isn't inlined, the compiler will generate a tail call here. But that'd cause the return address visible to dlsym to be foo's caller instead, which is not what we want if foo is called from a different object.